src/mgmt/cli_mgmt.c

```diff
@@ -417,6 +428,65 @@ get_attach_info(const char *name, int *npsrs, struct dc_mgmt_psr **psrs,
 	return rc;
 }
 
+/*
+ * Get the CaRT network configuration for this client node
+ * via the get_attach_info() dRPC.
+ * Configure the client's local environment with these parameters
+ */
+int dc_network_cfg(const char *name)
+{
+	int rc;
+	int npsrs;
+	char buf[SYS_INFO_BUF_SIZE];
+	struct sys_info sy_info;
+	struct dc_mgmt_psr *psrs;
+
+	if (name == NULL)
+		name = DAOS_DEFAULT_SYS_NAME;
+
+	/* Query the agent for the CaRT network configuration parameters */
+	rc = get_attach_info(name, &npsrs, &psrs, &sy_info);
+	if (rc != 0)
+		return rc;
+
+	/* These three are always set */
+	rc = setenv("CRT_PHY_ADDR_STR", sy_info.provider, 1);
+	if (rc != 0)
+		return errno;
+
+	sprintf(buf, "%d", sy_info.crt_ctx_share_addr);
+	rc = setenv("CRT_CTX_SHARE_ADDR", buf, 1);
+	if (rc != 0)
+		return errno;
+
+	sprintf(buf, "%d", sy_info.crt_timeout);
+	rc = setenv("CRT_TIMEOUT", buf, 1);
+	if (rc != 0)
+		return errno;
+
+	/* Allow client env overrides for these two */
```

Comment:
I'm a little wary of this, tbh. I'm concerned that people will have stale/incorrect environment and get mysterious failures. I think if we're going to allow these to be overridden by the user environment, we should at least emit a warning if the overriding values differ from the computed values.

Reply:
It would be good to get some other opinions to see if emitting a message describing the use of the user provided variables is deemed too spammy.  This does bring up an adjacent issue though.  These environment variables are not unset upon client termination.  I think it would make sense to unset at least OFI_INTERFACE and OFI_DOMAIN upon exit, if the auto-generated values were used.  That way, if another client is launched in the same shell, it will use the new values of OFI_INTERFACE and OFI_DOMAIN instead of what was written to the environment on the last run.

Reply:
Sure, other opinions wouldn't hurt. Btw, unless I'm mistaken, [setenv(3)](http://man7.org/linux/man-pages/man3/setenv.3.html) only modifies the process environment, not the shell environment. So there's no need to reset it.

Reply:
Just double-checked this:
```c
[mjmac@linuxvm tmp]$ cat setenv.c
#include <stdio.h>
#include <stdlib.h>

void main()
{
        char *env_name = "TEST_ENV";
        char *test_val = "TEST_VAL";

        printf("%s: %s\n", env_name, getenv(env_name));
        printf("updating %s\n", env_name);
        setenv(env_name, test_val, 1);
        printf("%s: %s\n", env_name, getenv(env_name));
}
[mjmac@linuxvm tmp]$ cc -o setenv setenv.c && TEST_ENV=foo ./setenv; echo "\$TEST_ENV: $TEST_ENV"
TEST_ENV: foo
updating TEST_ENV
TEST_ENV: TEST_VAL
$TEST_ENV:
```

Reply:
Thanks for the double check!  I appreciate the effort.

Reply:
While we see if there are other comments on emitting a warning if the user's environment variables conflict with the suggested interface and domain ... I wanted to add that if we emit such a message on conflict, it's one of those things that has potential to show output sporadically.  If there are 2 or more network device options available on that NUMA node, then the load balancer will provide a different device to use each time the get_attach_info() is called, which is supposed to be once per startup.  If the user variables agree with the answer on the first run, they may not agree on the second run, depending on other client activity on that node.  In the run where they agree, there would be no warning message.  Then on the next run, there could be.    It's not a harmful problem to emit the message, but it has potential to cause alarm until people understand what's happening.  It could be made more regular if we decided to always emit the configuration being used, and show the source of the data as coming from the user env, versus automatically provided by the agent.  That would remove any ambiguity, but comes at the expense of a line or two of output.

Reply:
I added D_INFO output that occurs when using the client provided overrides for OFI_INTERFACE or OFI_DOMAIN.